my-app/src/modules/About/ProductDetails.jsx

```diff
@@ -0,0 +1,87 @@
+import {useLocation} from 'react-router-dom';
+import {useState, useContext} from "react";
+import  ReactImageMagnify from "react-image-magnify";
+import { ToastContainer, toast } from 'react-toastify';
+import 'react-toastify/dist/ReactToastify.css';
+import { MyContext } from '../../data/MyContext';
+import Header from "../../Shared/Header/Header";
+import Footer from "../../Shared/Footer/Footer";
+import handleAddToCart from "./Services";
+import Category from "../../Shared/Category/Category";
+
+export default function ProductDetails() {
+  const location = useLocation();
+  const { product} = location.state;
+  const [quantity, setQuantity] = useState(1);
+  const { productQuantities, setProductQuantities } = useContext(MyContext);
+  const formattedPrice = product.price.toLocaleString();
+
+
+  const increaseQuantity = () => {
+    setQuantity(quantity + 1);
+  };
+
+  const decreaseQuantity = () => {
+    if (quantity > 1) {
+      setQuantity(quantity - 1);
+    }
+  };
+
+   const handleCartClick = async () => {
+    try {
+      await handleAddToCart(product, quantity);
```

Comment:
Use `async` `await` inside services.js, also don't use handle for services or utils functions.
It should be like `addToCart()`